src/Printer/Printer.php

```diff
@@ -97,6 +97,7 @@ final class Printer
 		ArrayShapeNode::class . '->items' => ', ',
 		ObjectShapeNode::class . '->items' => ', ',
 		CallableTypeNode::class . '->parameters' => ', ',
+		CallableTypeNode::class . '->templates' => ', ',
```

Comment:
1) This change is not tested.
2) I believe we need to test other situations. Like when gets `CallableTypeNode` added or removed, we also need to add `<...>` and remove `<...>` around them.

Reply:
what test should I use as a base to start?

Reply:
All the tests in PrinterTest, for example:

```php
		yield [
			'/** @var string&(integer|float) */',
			'/** @var string&(int|float) */',
			new class extends AbstractNodeVisitor {

				public function enterNode(Node $node)
				{
					if ($node instanceof IdentifierTypeNode && $node->name === 'integer') {
						$node->name = 'int';
					}

					return $node;
				}

			},
		];
```

This test accepts an original PHPDoc, modifies it using the visitor, and then asserts that the printer correctly prints the modified PHPDoc.

Reply:
> This change is not tested.

https://github.com/phpstan/phpdoc-parser/blob/1.24.2/src/Printer/Printer.php#L100 is not tested either, please add tests or please loosen this requirement for this contribution.

Reply:
> I believe we need to test other situations. Like when gets CallableTypeNode added or removed, we also need to add <...> and remove <...> around them.

addressed